test/framework/src/TxBuilder.cpp

```diff
@@ -154,14 +154,16 @@ TxBuilder& TxBuilder::AddPegoutKernel(const uint64_t amount, const uint64_t fee,
     return *this;
 }
 
-mw::Transaction::CPtr TxBuilder::Build()
+mw::Transaction::CPtr TxBuilder::Build(bool sort)
```

Comment:
I think it should always sort, since this is meant for building a transaction, not a whole mweb state.  Instead, we should change the KernelSumValidator::ValidateState test to build multiple blocks, the first one should do the peg-ins, and the final one should do the peg-outs.

Then we can also add a test where the first block does the peg-outs, and the final one does the peg-ins. In that scenario, we should expect a failure.